### PR TITLE
Lucas Theorem implementation

### PR DESCRIPTION
**Pull request by Rishikeshrajrxl**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [X] I've read the contribution guidelines.
- [X] I've checked the issue list before deciding what to submit.
- [X] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#1699 
(Write your answer here.)

### CODE REVIEW

**Comment by Rishikeshrajrxl** [comment]
@shraddhavp  can you please help me with these checks, why not successful.

**Comment by shraddhavp** [comment]
@Rishikeshrajrxl  dont worry it is a deepsource check fail,it is a new feature where implementation is still in progress ,it will soon be fixed ,it is not the error of PR dont worry

**Comment by Rishikeshrajrxl** [comment]
@atarax665, @shraddhavp please review this PR.

File: Python/math/Lucas Theorem.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                                1000C300 = (5C1) * (11C10) * (12C1)
+                                         = 5 * 11 * 12
+                                         = 5 *(-2) * (-1) 
+                                         = 10
+Example 2:
+Input:  n = 10, r = 2, p = 13
+Output: 6
+'''
+#defining the modulo function.
+def Mod(n, r, p):
```
Comment (atarax665) [inline Python/math/Lucas Theorem.py]: Add docstring to explain how this function works.

File: Python/math/Lucas Theorem.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    nth = int(n % p); 
+    rth = int(r % p);            
+    return (Lucas(int(n / p), int(r / p), p) * Mod(nth, rth, p)) % p; 
+########################################################################
+
+n = int(input("Enter The Value of n : "))
+r = int(input("Enter The Value of r : "))
+p = int(input("Enter The Value of p : "))
+print("Value of nCr % p is",Lucas(n, r, p)); 
+
```
Comment (atarax665) [inline Python/math/Lucas Theorem.py]: Remove all these blank lines.

File: Python/math/Lucas Theorem.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+             1000 = 5(13^2) + 11(13) + 12 
+              300 = 1(13^2) + 10(13) + 1
+        Then apply Lucas Theorem:
+                                1000C300 = (5C1) * (11C10) * (12C1)
+                                         = 5 * 11 * 12
+                                         = 5 *(-2) * (-1) 
+                                         = 10
+Example 2:
+Input:  n = 10, r = 2, p = 13
+Output: 6
```
Comment (atarax665) [inline Python/math/Lucas Theorem.py]: Add time and space complexity of the program.

**Review by atarax665 — CHANGES_REQUESTED** [review]
Add a comment that you have referred to [GFG article](https://www.geeksforgeeks.org/compute-ncr-p-set-2-lucas-theorem/) as the code looks quite similar.

**Comment by atarax665** [comment]
Please update Readme.md too.

**Comment by Rishikeshrajrxl** [comment]
@atarax665, @plazzy99, please have a look

**Review by vatsalkesarwani12 — APPROVED** [review]
👍🏻 
Code looks good to me 


I want to add something,
Contributing to an open-source project like this helps you to learn new things, upskill yourself. 

But if, if you are simply referring to the code from the websites, making few changes to it and submitting them. Then my friend you all are not learning anything and simply wasting the time of raising a pr and asking for a review.

**Comment by Rishikeshrajrxl** [comment]
> 👍🏻
> Code looks good to me
> 
> I want to add something,
> Contributing to an open-source project like this helps you to learn new things, upskill yourself.
> 
> But if, if you are simply referring to the code from the websites, making few changes to it and submitting them. Then my friend you all are not learning anything and simply wasting the time of raising a pr and asking for a review.

Thanks, @plazzy99  for your feedback, From next time I will keep that in mind and try to implement on my own. Thanks again.

File: Python/math/README.md [hunk 4]
```diff
@@ -1 +1 @@
-a
+[Lucas Theorem](https://github.com/Rishikeshrajrxl/NeoAlgo/blob/rishikeshrajrxl/Python/math/Lucas%20Theorem.py)
```
Comment (atarax665) [inline Python/math/README.md]: You need to update the main [README.md](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/README.md) of Python folder.

**Review by atarax665 — CHANGES_REQUESTED** [review]
LGTM, just update the correct README.md.

**Comment by atarax665** [comment]
Update [this](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/README.md) README.md, and revert all the changes made to the README.md of the Math sub-folder.

**Comment by Rishikeshrajrxl** [comment]
> Update [this](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/README.md) README.md, and revert all the changes made to the README.md of the Math sub-folder.

@atarax665  Done the changes

**Comment by Rishikeshrajrxl** [comment]
@atarax665, please review

**Comment by Rishikeshrajrxl** [comment]
@siddharth25pandey  @SKAUL05 , @Abhijit2505  please review this

**Review by atarax665 — APPROVED** [review]

**Pull request merged**